src/services/comparer.ts

```diff
@@ -1,18 +1,45 @@
-import { commands, Uri, extensions, window } from 'vscode';
+import { commands, Uri } from 'vscode';
 import { compare, fileCompareHandlers,type Difference } from 'dir-compare';
 import { openFolder } from './openFolder';
 import * as path from 'path';
 import { DiffViewTitle, getConfiguration } from './configuration';
 import { pathContext } from '../context/path';
-import { compareIgnoredExtension, compareName, validate } from './ignoreExtensionTools';
+import { compareName, validate } from './ignoreExtensionTools';
 import { CompareOptions, type DiffPathss, type ViewOnlyPaths } from '../types';
 import { log, printOptions, printResult } from './logger';
 import { showErrorMessage } from '../utils/ui';
 import { validatePermissions } from './validators';
 import { getIncludeAndExcludePaths } from './includeExcludeFilesGetter';
 import { getGitignoreFilter } from './gitignoreFilter';
+import { FileParserService } from './fileParser';
+import { ParsedDiffViewer } from './parsedDiffViewer';
+
+let fileParserService: FileParserService | null = null;
+let parsedDiffViewer: ParsedDiffViewer | null = null;
+
+function getFileParserService(): FileParserService {
+  if (!fileParserService) {
+    fileParserService = new FileParserService();
+  }
+  return fileParserService;
+}
+
+function getParsedDiffViewer(): ParsedDiffViewer {
+  if (!parsedDiffViewer) {
+    parsedDiffViewer = new ParsedDiffViewer(getFileParserService());
+  }
+  return parsedDiffViewer;
+}
+
+/**
+ * Clean up resources when the extension is deactivated
+ */
```

Comment:
Also, code comments are not needed.. unless they are needed 🙏  (same for other places)